fastcrawler/test_utils/test_server.py

```diff
@@ -0,0 +1,64 @@
+from typing import Optional, Callable
+
+from fastcrawler.engine.contracts import Request, Response
+from fastcrawler.test_utils.routes import Route, NoMatchFound
+
+
+class TestServer:
+    """A class that represents a test server.
+
+    This class can store a list of routes and return a response for a given request.
+
+    Attributes:
+        routes (list[Route]): A list of routes that the server can handle.
+    """
+
+    def __init__(self, routes: Optional[list[Route]] = None) -> None:
+        self.routes = routes or []
+
+    def add_route(self, route: Route) -> None:
+        """Add a route to the server.
+
+        Args:
+            route (Route): The route to add.
+        """
+        self.routes.append(route)
+
+    def get_response(self, request: Request) -> Callable[..., Response]:
+        """Get a response for a given request.
+
+        Args:
+            request (Request): The request to handle.
+
+        Returns:
+            Response: The response function from the matching route.
+
+        Raises:
+            NoMatchFound: If no route matches the request.
+        """
+        assert request.method
+        response_function = self._find_response_function(request.url, request.method)
+
+        if response_function is None:
+            raise NoMatchFound(request.url, {})
+        # Otherwise, return the response from the route
+        return response_function
+
+    def _find_response_function(self, url: str, method: str) -> Callable[..., Response] | None:
+        """Find a matching route for a given url and method.
+
+        Args:
+            url (str): The url to match.
+            method (str): The method to match.
+
+        Returns:
+            Optional[Route]: The matching route or None if not found.
+        """
+        # Loop through the routes and return the first response that matches
+        for route in self.routes:
+            try:
+                return route.url_path_for(url, method)
+            except NoMatchFound:
+                pass
```

Comment:
rework this with first_true :)
from more_itertools library.

Reply:
it's forces the new dependency, we have three way:
1. use this without any change
2. use more_itertools and force new dependency
3. rewrite first_true in this project

Reply:
more_itertools is not a "heavy" dependency, 
I'd argue we can have it, we may need it in future as well.

Reply:
The `more_itertools` is added!